SocialCareCaseViewerApi/V1/Controllers/CaseStatusController.cs

```diff
@@ -100,5 +101,44 @@ public IActionResult CreateCaseStatus([FromBody] CreateCaseStatusRequest request
                 return BadRequest(e.Message);
             }
         }
+
+        /// <summary>
+        /// Edit a Case Status
+        /// </summary>
+        /// <param name="request"></param>
+        /// <param name="caseStatusId"></param>
+        /// <response code="201">Successfully updated a case status</response>
+        /// <response code="400">Invalid request received</response>
+        [ProducesResponseType(typeof(CaseStatus), 200)]
+        [HttpPatch]
+        [Route("residents/case-statuses/{caseStatusId:long}")]
+        public IActionResult UpdateCaseStatus([FromRoute] long caseStatusId, [FromBody] UpdateCaseStatusRequest request)
+        {
+            var validator = new UpdateCaseStatusValidator();
+            var validationResults = validator.Validate(request);
+
+            if (!validationResults.IsValid)
+            {
+                return BadRequest(validationResults.ToString());
+            }
+
+            try
+            {
+                var caseStatus = _caseStatusesUseCase.ExecuteUpdate(caseStatusId, request);
+                return Ok(caseStatus);
+            }
+            catch (Exception e) when (
+                e is PersonNotFoundException ||
+                e is WorkerNotFoundException ||
```

Comment:
Would the Not found exceptions be `404` s? (I don't have strong opinions on this, just interested in learning about status codes in general 🙂 )

Reply:
I consider if the worker/resident isn't found then the request to update the CaseStatus was a Badrequest. 
The resident/worker is not the primary focus of the request so I only return a 404 when the actual CaseStatus isn't found.

Reply:
That makes sense. 🙌🏾 . Thanks!